app/api/answers/get_best.go

```diff
@@ -62,35 +65,25 @@ func (srv *Service) getBestAnswer(rw http.ResponseWriter, httpReq *http.Request)
 	var result []map[string]interface{}
 
 	var bestAnswerQuery *database.DB
-	if ok {
-		// check 'can_watch'>='answer' permission on the answers.item_id
-		itemPerms := store.Permissions().MatchingUserAncestors(user).
-			WherePermissionIsAtLeast("watch", "answer").
-			Where("permissions.item_id = answers.item_id").
-			Select("1").
-			Limit(1)
+	if watchedGroupOK {
+		// the following checks were made by ResolveWatchedGroupID:
+		// - watched_group_id must be a participant
+		// - the current user is able to watch the participant
 
-		// check if able to watch the participant
-		participantPerms := store.ActiveGroupAncestors().ManagedByUser(user).
-			Where("groups_ancestors_active.child_group_id = answers.participant_id").
-			Where("can_watch_members").
-			Select("1").
-			Limit(1)
+		// check 'can_watch'>='answer' permission on the answers.item_id
+		if !user.CanWatchItemAnswer(store, itemID) {
+			return service.InsufficientAccessRightsError
+		}
 
 		bestAnswerQuery = store.Answers().
-			Where(`?`, itemPerms.SubQuery()).
-			Where(`?`, participantPerms.SubQuery()).
 			Where("participant_id = ?", watchedGroupID)
 	} else {
 		// check 'can_view'>='content' permission on the answers.item_id
-		itemPerms := store.Permissions().MatchingUserAncestors(user).
-			WherePermissionIsAtLeast("view", "content").
-			Where("permissions.item_id = answers.item_id").
-			Select("1").
-			Limit(1)
+		if !user.CanViewItemContent(store, itemID) {
+			return service.InsufficientAccessRightsError
+		}
```

Comment:
By https://france-ioi.github.io/algorea-devdoc/design/access-rights/items/#changing-and-propagating-permissions, a user cannot have `can_watch>=answer` without having `can_view ≥ content` so this check is useless. 
(anyway, this kind of checks should have been done in one single sql request)

Reply:
That good to know! In this case though the can_view>= is checked independently in else {} block. Is this comment a mistake or am I missing something?